cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,290 @@
+
+An overview diagram of `preprogrammed` mode is below. A diagram of `orchestrated` mode
+is provided in [an appendix](#orchestrated-mode).
+
+![Overview of `preprogrammed` mode](pictures/FPGA-preprogrammed.png)
+
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+Components can generally be installed either using DaemonSets or running them
+'by hand' on each node.
+
+## Pre-built images
+
+Pre-built images of the components are available on the [Docker hub](https://hub.docker.com/u/intel).
+These images are automatically built and uploaded to the hub from the latest `master` branch of
+this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers (of the form `x.y.z`, matching the branch/tag release number in this repo).
+
+The deployment YAML files supplied with these components in this repository use the images with the
+`devel` tag by default. If you do not build your own local images, then your Kubernetes cluster may
+pull down the `devel` images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the YAML deployment files appropriately.
+
+The following images are available on the Docker hub:
+
+- [The FPGA plugin](https://hub.docker.com/r/intel/intel-fpga-plugin)
+- [The FPGA admisson webhook](https://hub.docker.com/r/intel/intel-fpga-admissionwebhook)
+- [The FPGA CRI-O plugin (in the `initcontainer` image)](https://hub.docker.com/r/intel/intel-fpga-initcontainer)
+
+## Dependencies
+
+All components have the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about)
+
+To obtain a fully operational FPGA enabled cluster, you must install all three
+major components:
+
+-   [FPGA device plugin](README.md) (this component)
 -   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
 -   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
 
+The CRI-O hook is only *required* if `orchestrated` FPGA bitstream programming mode is
+being used, but is installed by default by the
+[FPGA plugin DaemonSet YAML](../../deployments/fpga_plugin/fpga_plugin.yaml), and is benign
+in `preprogrammed` mode.
 
-### Get source code:
-```
-    $ mkdir -p $GOPATH/src/github.com/intel/
-    $ cd $GOPATH/src/github.com/intel/
-    $ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
-```
+If using the `preprogrammed` mode, and therefore *not* using the
+CRI-O prestart hook, runtimes other than CRI-O can be used (that is, the CRI-O hook presently
+*only* works with the CRI-O runtime).
 
-### Build FPGA device plugin:
-```
-    $ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-    $ make fpga_plugin
-```
+The FPGA device plugin requires a Linux Kernel FPGA driver to be installed and enabled to
+operate. The plugin supports the use of either of two drivers:
 
```

Comment:
It would make sense to mention that plugin detects with driver is loaded, so no additional configuration is needed.

Reply:
Done